redash/query_runner/exasol.py

```diff
@@ -0,0 +1,153 @@
+import json
+import datetime
+
+from redash.query_runner import *
+
+def _exasol_type_mapper(val, data_type):
+    if val is None:
+        return None
+    elif data_type['type'] == 'DECIMAL':
+        if data_type['scale'] == 0 and data_type['precision'] < 16:
+            return int(val)
+        elif data_type['scale'] == 0 and data_type['precision'] >= 16:
+            return val
+        else:
+            return float(val)
+    elif data_type['type'] == 'DATE':
+        return datetime.date(int(val[0:4]), int(val[5:7]), int(val[8:10]))
+    elif data_type['type'] == 'TIMESTAMP':
+        return datetime.datetime(int(val[0:4]), int(val[5:7]), int(val[8:10]),           # year, month, day
+                                 int(val[11:13]), int(val[14:16]), int(val[17:19]),      # hour, minute, second
+                                 int(val[20:26].ljust(6, '0')) if len(val) > 20 else 0)  # microseconds (if available)
+    else:
+        return val
+
+def _type_mapper(data_type):
+    if data_type['type'] == 'DECIMAL':
+        if data_type['scale'] == 0 and data_type['precision'] < 16:
+            return TYPE_INTEGER
+        elif data_type['scale'] == 0 and data_type['precision'] >= 16:
+            return TYPE_STRING
+        else:
+            return TYPE_FLOAT
+    elif data_type['type'] == 'DATE':
+        return TYPE_DATE
+    elif data_type['type'] == 'TIMESTAMP':
+        return TYPE_DATETIME
+    else:
+        return TYPE_STRING
+
+try:
+    import pyexasol
+    enabled = True
+except ImportError:
+    enabled = False
+
+
+class Exasol(BaseQueryRunner):
+    noop_query = "SELECT 1 FROM DUAL"
+
+    def test_connection(self):
+        data, error = self.run_query(self.noop_query, None)
+        if error is not None:
+            raise Exception(error)
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": {
+                    "type": "string"
+                },
+                "password": {
+                    "type": "string"
+                },
+                "host": {
+                    "type": "string",
+                    "default": "127.0.0.1"
```

Comment:
As it's very unlikely to have an Exasol instance running on the same host as Redash, I would remove this default value. Also let's make `host` a required value.